src/main/java/org/ice4j/message/ChannelData.java

```diff
@@ -196,7 +196,12 @@ public static ChannelData decode(byte binMessage[], char offset, char arrayLen)
         }
 
         msgLen = (char)((binMessage[offset++]<<8) | (binMessage[offset++]&0xFF));
-        if(msgLen != (binMessage.length - 4))
+        /*
+         * message can be padded
+         * https://tools.ietf.org/html/rfc5766#section-11.5
+         */
+        if (msgLen != (binMessage.length - HEADER_LENGTH)
+            && ((msgLen % 4) + 4) != (binMessage.length - HEADER_LENGTH))
```

Comment:
I might be tired and don't know ICE or TURN, but I'd assume that `binMessage.length - HEADER_LENGTH` can be legally greater than `[4,7]`.